src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt

```diff
@@ -1,214 +1,144 @@
 
-    // Chat panel
-    messagesPanel.layout = VerticalFlowLayout(VerticalFlowLayout.TOP, 0, 0, true, true)
-    val chatPanel = ChatScrollPane(messagesPanel)
+  private val subscriptionPanel = SubscriptionTabPanel()
 
-    // Controls panel
-    sendButton = createSendButton()
-    promptPanel =
-        PromptPanel(
-            chatMessageHistory,
-            ::sendChatMessage,
-            sendButton,
-            isGenerating = stopGeneratingButton::isVisible)
-    val stopGeneratingButtonPanel = JPanel(FlowLayout(FlowLayout.CENTER, 0, 5))
-    stopGeneratingButtonPanel.minimumSize = Dimension(Short.MAX_VALUE.toInt(), 0)
-    stopGeneratingButton.addActionListener {
-      inProgressChat.abort()
-      stopGeneratingButton.isVisible = false
-      sendButton.isEnabled = promptPanel.textArea.text.isNotBlank()
-      ensureBlinkingCursorIsNotDisplayed()
-      commandsPanel.enableAllButtons()
+  init {
+    CodyAgentService.getInstance(project).addStartupAction { agent ->
+      agent.client.onChatUpdate = Consumer { params ->
+        chatPanels[params.id]?.let { Chat.processResponse(project, it, params.message) }
+      }
     }
 
-    stopGeneratingButton.isVisible = false
-    stopGeneratingButtonPanel.add(stopGeneratingButton)
-    stopGeneratingButtonPanel.isOpaque = false
-    contextView = EnhancedContextPanel(project)
-    val lowerPanel = LowerPanel(stopGeneratingButtonPanel, promptPanel, contextView)
+    tabbedPane.insertSimpleTab("Chat", chatContainerPanel, CHAT_TAB_INDEX)
+    tabbedPane.insertSimpleTab("Chat History", historyTree, HISTORY_TAB_INDEX)
+    tabbedPane.insertSimpleTab("Commands", commandsPanel, RECIPES_TAB_INDEX)
+    tabbedPane.insertSimpleTab("Subscription", subscriptionPanel, SUBSCRIPTION_TAB_INDEX)
+
+    tabbedPane.addChangeListener {
+      if (tabbedPane.selectedIndex == CHAT_TAB_INDEX) {
+        val currentChatPanel = chatContainerPanel.getComponent(0) as? ChatPanel
+        currentChatPanel?.requestFocusOnInputPrompt()
+      }
+    }
 
-    // Main content panel
-    contentPanel.layout = BorderLayout(0, 0)
-    contentPanel.border = BorderFactory.createEmptyBorder(0, 0, 10, 0)
-    contentPanel.add(chatPanel, BorderLayout.CENTER)
-    contentPanel.add(lowerPanel, BorderLayout.SOUTH)
-    tabbedPane.addChangeListener { focusPromptInput() }
-    val singInWithSourcegraphPanel = SignInWithSourcegraphPanel(project)
     allContentPanel.add(tabbedPane, "tabbedPane", CHAT_PANEL_INDEX)
-    allContentPanel.add(
-        singInWithSourcegraphPanel, SING_IN_WITH_SOURCEGRAPH_PANEL, SIGN_IN_PANEL_INDEX)
-    allContentLayout.show(allContentPanel, SING_IN_WITH_SOURCEGRAPH_PANEL)
-    refreshPanelsVisibility()
+    allContentPanel.add(signInWithSourcegraphPanel, SIGN_IN_PANEL, SIGN_IN_PANEL_INDEX)
+    allContentLayout.show(allContentPanel, SIGN_IN_PANEL)
 
+    ApplicationManager.getApplication().invokeLater { refreshPanelsVisibility() }
```

Comment:
I guess we are in the dispatched thread already so `invokeLater` is redundant (am I right?)